protobuf/src/stream.rs

```diff
@@ -787,31 +726,15 @@ impl<'a> CodedOutputStream<'a> {
     }
 
     pub fn write_raw_varint32(&mut self, value: u32) -> ProtobufResult<()> {
-        if self.buffer.len() - self.position >= 5 {
-            // fast path
-            let len = varint::encode_varint32(value, &mut self.buffer[self.position..]);
-            self.position += len;
-            Ok(())
-        } else {
-            // slow path
-            let buf = &mut [0u8; 5];
-            let len = varint::encode_varint32(value, buf);
-            self.write_raw_bytes(&buf[..len])
-        }
+        let buf = &mut [0u8; 5];
+        let len = varint::encode_varint32(value, buf);
+        self.write_raw_bytes(&buf[..len])
     }
 
     pub fn write_raw_varint64(&mut self, value: u64) -> ProtobufResult<()> {
-        if self.buffer.len() - self.position >= 10 {
-            // fast path
-            let len = varint::encode_varint64(value, &mut self.buffer[self.position..]);
-            self.position += len;
-            Ok(())
-        } else {
-            // slow path
-            let buf = &mut [0u8; 10];
-            let len = varint::encode_varint64(value, buf);
-            self.write_raw_bytes(&buf[..len])
-        }
+        let buf = &mut [0u8; 10];
+        let len = varint::encode_varint64(value, buf);
+        self.write_raw_bytes(&buf[..len])
```

Comment:
Writing into an intermediate buffer slows down serialization. Try to run `bench_write_raw_varint_32` I've just committed with this change: you will notice a significant slowdown.